src/primitive.h

```diff
@@ -951,7 +951,7 @@ Object* get_absolute_path(Process* process, const wchar_t* pathname, wchar_t* ou
   Object* _raw_##name = __args[-(N)];                                        \
   uninitialized_t _u_##name;                                                 \
   Blob name(_u_##name);                                                      \
-  if (!_raw_##name->byte_content(process->program(), &name, STRINGS_OR_BYTE_ARRAYS)) FAIL(WRONG_OBJECT_TYPE);
+  if (!_raw_##name->byte_content(process->program(), &name, STRINGS_OR_BYTE_ARRAYS)) FAIL(WRONG_BYTES_TYPE);
```

Comment:
So here BYTES / io.Data is the same thing -- or is BYTES a subset of io.Data? It feels like it would be nice if the error message made it more recognizable that this was the wrong kind of io.Data when that is the case.

I am all for cleaning this up further when we look at the exception hierarchy.

Reply:
I agree. This should be cleaned up a bit.
In theory, the Toit type check should guarantee that only `io.Data` reaches the primitive functions.
Every primitive function that takes an `io.Data` should then deal with the `WRONG_BYTES_TYPE` immediately and convert the arguments to ByteArrays (potentially in a chunked way).
This means that the user should never see the `WRONG_BYTES_TYPE`. If we have implemented our Toit functions correctly, then it should always be used as a signal to convert, and never as an exception.